### PR TITLE
Edit final block

### PR DESCRIPTION
**Pull request by haisenCx**
Optimized the SQL efficiency of LastFinalizedCompact function

### CODE REVIEW

File: core/blockchain/blockchain.go [hunk 1]
```diff
@@ -367,14 +368,15 @@ func (bc *BlockChain) LastFinalizedCompact() (*CompactBlock, error) {
...
-	}).Order("height").Find(&bss).Error
+	var bs BlocksScheme
+	err := bc.chain.Db().Model(&BlocksScheme{}).
+		Where("finalize = ?", true).
+		Order("height DESC").
+		Limit(1).
+		First(&bs).Error
 	if err != nil {
-		return nil, err
+		return nil, nil
```
Comment (Lawliet-Chan) [inline core/blockchain/blockchain.go:378]:
```suggestion
		return nil, err
```

File: apps/poa/poa.go [hunk 2]
```diff
@@ -266,11 +266,13 @@ func (h *Poa) EndBlock(block *types.Block) {
...
 	//logrus.WithField("block-height", block.Height).WithField("block-hash", block.Hash.String()).
 	//	Info("finalize block")
-
 	if h.cfg.PrettyLog {
 		log.DoubleLineConsole.Info(fmt.Sprintf("finalize block, height=%d, hash=%s", block.Height, block.Hash.String()))
 	}
-	h.Chain.Finalize(block)
+	err := h.Chain.Finalize(block)
+	if err != nil {
+		logrus.Info("finalize block failed: %s", err)
```
Comment (Lawliet-Chan) [inline apps/poa/poa.go:274]:
```suggestion
		logrus.Error("finalize block failed: ", err)
```

File: core/blockchain/blockchain.go [hunk 3]
```diff
@@ -367,14 +368,15 @@ func (bc *BlockChain) LastFinalizedCompact() (*CompactBlock, error) {
...
-	var bss []BlocksScheme
-	err := bc.chain.Db().Model(&BlocksScheme{}).Where(&BlocksScheme{
-		Finalize: true,
-	}).Order("height").Find(&bss).Error
+	var bs BlocksScheme
+	err := bc.chain.Db().Model(&BlocksScheme{}).
+		Where("finalize = ?", true).
+		Order("height DESC").
+		Limit(1).
+		First(&bs).Error
```
Comment (Lawliet-Chan) [inline core/blockchain/blockchain.go:376]:
```suggestion
		Find(&bs).Error
```

Comment (haisenCx) [reply core/blockchain/blockchain.go]: fixed

File: apps/poa/poa.go (hunk 2 above)
Comment (haisenCx) [reply apps/poa/poa.go]: fixed

File: core/blockchain/blockchain.go (hunk 1 above)
Comment (haisenCx) [reply core/blockchain/blockchain.go]: fixed

**Pull request merged**